File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.py
import math

def solution(W, H):
    # 총 격자칸 개수
    total_squares = W * H
    
    # 대각선이 지나가는 격자칸 개수
    diagonal_squares = W + H - math.gcd(W, H)
    
    # 사용할 수 있는 격자칸 개수
    usable_squares = total_squares - diagonal_squares
    
    return usable_squares

# 테스트
print(solution(8, 12))  # 출력: 80

File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.java
import java.math.BigInteger;

class Solution {
    public long solution(int W, int H) {
        // 최대공약수 계산
        int gcd = BigInteger.valueOf(W).gcd(BigInteger.valueOf(H)).intValue();

        // 총 격자 칸 수
        long totalSquares = (long) W * H;

        // 대각선이 지나가는 격자 칸 수
        long diagonalSquares = (long) W + H - gcd;

        // 사용할 수 있는 격자 칸 수
        return totalSquares - diagonalSquares;
    }
}